app/hydrator/controllers/detail/canvas-ctrl.js

```diff
@@ -65,6 +65,9 @@ angular.module(PKG.name + '.feature.hydrator')
     };
 
     this.setActiveNode = function() {
+      // need to use new config and nodes
+      var pipelineConfig = this.PipelineDetailStore.getState().config;
+      var nodes = this.HydratorPlusPlusHydratorService.getNodesFromStages(pipelineConfig.stages);
```

Comment:
Nit: this shadows the `nodes` variable instantiated on L54. Should this variable have a different name? Or, should the variable in the outer scope be updated?

Reply:
I believe it's better to update them in outer scope, if there is a way